files/image_config/process-reboot-cause/process-reboot-cause.service

```diff
@@ -1,6 +1,6 @@
 [Unit]
 Description=Reboot cause determination service
-After=rc-local.service
+After=pmon.service
```

Comment:
Why change this?

Reply:
Hi Joe, 
The hardware drivers required to provide reboot reason are installed only inside platform-modules-<device>  service. 

When process-reboot-cause service starts before the pmon service, loading of drivers are not complete and it leads to race condition where the show reboot-cause CLI return incorrect answers.

If process-reboot-cause loads after pmon.service, the platform drivers would have successfully loaded by then and show reboot-cause cli will return correct value

Thanks
Sridhar.R

Reply:
Hi Sidhar,

SONiC requires that hardware reboot reason and watchdog functionality need to be available in the host OS. The sonic_platform package should be installed in both the host OS and PMon container. Host OS services should not have any dependencies on the containers. We want reboot cause and watchdog to be as reliable as possible. Thanks.

Reply:
Hi Joe,
Thanks for the input. I have removed the pmon dependency from process-restart-cause service. 
Thanks
Sridhar.R